src/redis-cli.c

```diff
@@ -7726,40 +7739,62 @@ unsigned long long sendSync(redisContext *c, char *out_eof) {
         fprintf(stderr, "SYNC with master failed: %s\n", buf);
         exit(1);
     }
+
+    /* Handling PSYNC responses. */
+    if (!strncmp(buf, "+FULLRESYNC", 11) ||
+        !strncmp(buf, "+CONTINUE", 9))
+    {
+        int sync_partial = !strncmp(buf, "+CONTINUE", 9);
+        fprintf(stderr, "PSYNC replied %s\n", buf);
+        p = buf;
+        while(1) {
+            nread = readConn(c,p,1);
```

Comment:
that block is for reading the bulk size in case of FULLSYNC, why did do we get here in CONTINUE too?

Reply:
we send the PSYNC in the above, the `cliWriteConn(c, "", 0)` one, so it might get the CONTINUE in the response, and i handle it in here

Reply:
> but with CONTINUE isn't the next thing we get is the command stream?

that is true, the while 1 is actually needed, there are a few `\n` we need to read.

> btw, if the user declared REPLCONF CAPA PSYNC2, we may need to parse a more complicated +CONTINUE.
your code seems to already handle that, right?

yes, it can handle that

Reply:
ok, so let's put a comment as to what we're reading. i.e. "either payload size / EOF, or newlines after CONTINUE"